xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.xtable.conversion.ConversionSourceProvider;
+import org.apache.xtable.conversion.SourceTable;
+import org.apache.xtable.conversion.TargetTable;
+import org.apache.xtable.delta.DeltaConversionSourceProvider;
+import org.apache.xtable.hudi.HudiConversionSourceProvider;
+import org.apache.xtable.iceberg.IcebergConversionSourceProvider;
+import org.apache.xtable.service.models.ConvertTableRequest;
+import org.apache.xtable.service.models.ConvertTableResponse;
+import org.apache.xtable.service.models.RestTargetTable;
+import org.apache.xtable.service.spark.SparkHolder;
+import org.apache.xtable.service.utils.DeltaMetadataUtil;
+import org.apache.xtable.service.utils.HudiMedataUtil;
+import org.apache.xtable.service.utils.IcebergMetadataUtil;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class ConversionService {
+  private final SparkHolder sparkHolder;
+  private final ConversionControllerFactory controllerFactory;
+  private final IcebergMetadataUtil icebergUtil;
+  private final HudiMedataUtil hudiUtil;
+  private final DeltaMetadataUtil deltaUtil;
+
+  @Inject
+  public ConversionService(
+      SparkHolder sparkHolder,
+      ConversionControllerFactory controllerFactory,
+      IcebergMetadataUtil icebergUtil,
+      HudiMedataUtil hudiUtil,
+      DeltaMetadataUtil deltaUtil) {
+    this.sparkHolder = sparkHolder;
+    this.controllerFactory = controllerFactory;
+    this.icebergUtil = icebergUtil;
+    this.hudiUtil = hudiUtil;
+    this.deltaUtil = deltaUtil;
+  }
+
+  public ConvertTableResponse convertTable(ConvertTableRequest request) {
+    Configuration conf = sparkHolder.jsc().hadoopConfiguration();
+    ConversionController conversionController = controllerFactory.create(conf);
+
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(request.getSourceTableName())
+            .basePath(request.getSourceTablePath())
+            .formatName(request.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : request.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(request.getSourceTableName())
+              .basePath(request.getSourceTablePath())
+              .formatName(targetFormat)
+              .build();
+      targetTables.add(targetTable);
+    }
+    ConversionConfig conversionConfig =
+        ConversionConfig.builder().sourceTable(sourceTable).targetTables(targetTables).build();
+    ConversionSourceProvider<?> conversionSourceProvider =
+        getConversionSourceProvider(request.getSourceFormat());
+    conversionController.sync(conversionConfig, conversionSourceProvider);
+
+    List<RestTargetTable> restTargetTables = new ArrayList<>();
+    for (String targetFormat : request.getTargetFormats()) {
+      if (targetFormat.equals("ICEBERG")) {
```

Comment:
Use TableFormat.ICEBERG constant available in the repo.

Reply:
ack